x/auth/types/client_tx_test.go

```diff
@@ -78,23 +77,29 @@ func TestStdTxBuilder_SetFeeAmount(t *testing.T) {
 	}
 	tx := stdTxBuilder.GetTx()
 	stdTxBuilder.SetFeeAmount(feeAmount)
+	feeTx := stdTxBuilder.GetTx().(sdk.FeeTx)
 	require.NotEqual(t, tx, stdTxBuilder.GetTx())
+	require.Equal(t, feeTx.GetFee(), feeAmount)
 	require.False(t, reflect.DeepEqual(tx, stdTxBuilder.GetTx()))
 }
 
 func TestStdTxBuilder_SetGasLimit(t *testing.T) {
 	stdTxBuilder, _ := setupStdTxBuilderTest(t)
 	tx := stdTxBuilder.GetTx()
 	stdTxBuilder.SetGasLimit(300000)
+	feeTx := stdTxBuilder.GetTx().(sdk.FeeTx)
 	require.NotEqual(t, tx, stdTxBuilder.GetTx())
```

Comment:
I think they might actually be equal because `TxBuilder` may be doing things by reference. All we need to do is check that the gas limit is properly set.
```suggestion
```

Reply:
They are different but we can just test `GasLimit`.